File: hexagonal-app-load-testing/app-helidon-java/src/main/java/app/config/Configuration.java
package app.config;

import business.common.KafkaAdapter;
import business.common.MongoDbAdapter;
import business.common.PrometheusAdapter;
import business.external.MonitoringService;
import business.external.PersistenceRepository;
import business.external.StreamService;
import business.service.CpuBoundUseCase;
import business.service.IoBoundUseCase;

import javax.enterprise.inject.Produces;
import javax.ws.rs.ext.Provider;

@Provider
public class Configuration {

    @Produces
    public PersistenceRepository persistenceRepository() {
        return new MongoDbAdapter();
    }

    @Produces
    public StreamService streamService() {
        return new KafkaAdapter();
    }

    @Produces
    public MonitoringService monitoringService() {
        return new PrometheusAdapter();
    }

    @Produces
    public IoBoundUseCase ioBoundUseCase(MonitoringService monitoringService, PersistenceRepository persistenceRepository) {
        return new IoBoundUseCase(monitoringService, persistenceRepository);
    }

    @Produces
    public CpuBoundUseCase cpuBoundUseCase(MonitoringService monitoringService, StreamService streamService) {
        return new CpuBoundUseCase(monitoringService, streamService);
    }
}

File: hexagonal-app-load-testing/app-spring-kotlin/src/main/kotlin/app/config/Configuration.kt
package app.config

import business.external.MonitoringService
import business.external.PersistenceRepository
import business.external.StreamService
import business.service.CpuBoundUseCase
import business.service.IoBoundUseCase
import business.common.KafkaAdapter
import business.common.MongoDbAdapter
import business.common.PrometheusAdapter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class Configuration {

    @Bean
    fun kafkaAdapter(): StreamService {
        return KafkaAdapter()
    }

    @Bean
    fun mongoDbAdapter(): PersistenceRepository {
        return MongoDbAdapter()
    }

    @Bean
    fun prometheusAdapter(): MonitoringService {
        return PrometheusAdapter()
    }

    @Bean
    fun cpuUseCase(monitoringService: MonitoringService, streamService: StreamService): CpuBoundUseCase {
        return CpuBoundUseCase(monitoringService, streamService)
    }

    @Bean
    fun ioUseCase(monitoringService: MonitoringService, persistenceRepository: PersistenceRepository): IoBoundUseCase {
        return IoBoundUseCase(monitoringService, persistenceRepository)
    }
}